### ISSUE TITLE
Observe items.* using updateSizeForItem

### ISSUE THREAD

**Issue by indolering** [PolymerElements/iron-list#260]
I've been manually calling updateSizeForItem after calling `this.set("path", value)`, but it would be preferable if `iron-list` observed items and updated their size automatically.  I can make this PR myself, but I have a couple of questions.

Would it be enough to just add a observer of should I be check to see if the element is rendered in the physical list?  

Also, would altering the size of items cause the view to jerk?  If so we could make this behavior configurable so that it can be disabled....

**Comment by blasten** [comment]
I agree. It should call `updateSizeForItem` automatically. `_forwardItemPath` should be the right spot. 
```js
this.updateSizeForItem(el._templateInstance[this.as]);
```

Feel free to send a PR!

**Comment by indolering** [comment]
Ahh, I have no idea what that code is doing, I'm not sure where to even c/p the line you gave me : p

**Comment by indolering** [comment]
Is this correct or do I need to worry about that path finding code?

```
      inst = el._templateInstance;

      if (inst.__key__ !== key) {
        return;
      }

      this.updateSizeForItem(el._templateInstance[this.as]);

      if (dot >= 0) {
        path = this.as + '.' + path.substring(dot+1);
        inst.notifyPath(path, value, true);
      } else {
        inst[this.as] = value;
      }
```

**Comment by blasten** [comment]
Now that I'm thinking this a little bit more, maybe the list shouldn't do that at all. Here's why: many of changes to the item won't necessarily change the size of the physical item. I agree that calling `updateSizeForItem` automatically makes things easier when the size is actually changing, but at the expense of calling `updateSizeForItem ` for the property changes that won't affect the size.  Let's keep it as it's right now. The user needs to call: 
```js
list.set('items.0.name', 'this might overflow');
list.updateSizeForItem();
```

**Comment by indolering** [comment]
But is there any real downside to calling it whenever the item changes?  It's *slightly* less efficient, but is overhead a real concern here?

If we go this route, then we need to reopen #259.  However, I just noticed that I'm still getting edge cases that the 50ms pad doesn't fix.
